src/test/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilderTest.java

```diff
@@ -0,0 +1,304 @@
+package bio.terra.service.snapshotbuilder.utils;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.equalToCompressingWhiteSpace;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import bio.terra.common.category.Unit;
+import bio.terra.common.exception.BadRequestException;
+import bio.terra.grammar.azure.SynapseVisitor;
+import bio.terra.model.SnapshotBuilderCriteria;
+import bio.terra.model.SnapshotBuilderCriteriaGroup;
+import bio.terra.model.SnapshotBuilderDomainCriteria;
+import bio.terra.model.SnapshotBuilderProgramDataListCriteria;
+import bio.terra.model.SnapshotBuilderProgramDataRangeCriteria;
+import bio.terra.service.snapshotbuilder.query.FilterVariable;
+import bio.terra.service.snapshotbuilder.query.Query;
+import bio.terra.service.snapshotbuilder.query.TablePointer;
+import bio.terra.service.snapshotbuilder.query.TableVariable;
+import java.math.BigDecimal;
+import java.util.List;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@Tag(Unit.TAG)
+class CriteriaQueryBuilderTest {
+
+  @Test
+  void generateRangeCriteriaFilterProducesCorrectSql() {
+    SnapshotBuilderProgramDataRangeCriteria rangeCriteria = generateRangeCriteria();
+    FilterVariable filterVariable =
+        new CriteriaQueryBuilder("person", SynapseVisitor.azureTableName("source_dataset_name"))
+            .generateFilterForRangeCriteria(rangeCriteria);
+
+    // Table name is null because there is no alias generated until it is rendered as a full query
+    assertThat(
+        "The sql generated is correct",
+        filterVariable.renderSQL(),
+        equalToCompressingWhiteSpace(
+            "(null.range_column_name >= 0 AND null.range_column_name <= 100)"));
+    // There are actually two entsries for "person" because the range filter generates two distinct
+    // filters which each reference person. Distinct is to fix this.
+    assertThat(
+        "The filter has only person within the table",
+        filterVariable.getTables().stream()
+            .map(TableVariable::getTablePointer)
+            .map(TablePointer::tableName)
+            .distinct()
+            .toList(),
+        equalTo(List.of("person")));
+  }
+
+  @Test
+  void generateListCriteriaFilterProducesCorrectSql() {
+    SnapshotBuilderProgramDataListCriteria listCriteria = generateListCriteria();
+    FilterVariable filterVariable =
+        new CriteriaQueryBuilder("person", SynapseVisitor.azureTableName("source_dataset_name"))
```

Comment:
If all tests use an instance of `CriteriaQueryBuilder()` you could create it in a `beforeEach()` method.